clap_complete/tests/testsuite/dynamic.rs

```diff
@@ -589,6 +589,45 @@ val3
         ]
     );
 }
+fn suggest_custom_valuehint() {
+    #[derive(Debug, Clone)]
+    struct ReadFileHint {
+        file: std::path::PathBuf,
```

Comment:
If we don't have to touch the filesystem as part of a test, let's not

Reply:
I want to add a case that accesses external resource to generate custom completion. Do you have any good advice?

Reply:
These use cases are independent of what functionality being tested.  All we care about is that its possible.  This is best left to examples.